lib/src/working_copy.rs

```diff
@@ -846,6 +858,13 @@ impl TreeState {
                             message: format!("Failed to stat file {}", entry.path().display()),
                             err,
                         })?;
+                        if metadata.len() > max_new_file_size {
+                            return Err(SnapshotError::NewFileTooLarge {
+                                path: entry.path().clone(),
+                                size: metadata.len(),
+                                max_size: max_new_file_size,
+                            });
+                        }
```

Comment:
It also applies to tracked files.

needs to test `maybe_current_file_state.is_none()`?

Reply:
Whoops, that was intended but got lost in a refactor.